### PR TITLE
Import REQUIRED_PACKAGES from userland-gate

### PR DESCRIPTION
**Pull request by xen0l**
This PR imports build dependency tracking from userland-gate.

### CODE REVIEW

File: make-rules/environment.mk [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	@/usr/sbin/psrinfo -vp
+	@/usr/sbin/ipadm show-addr
+	$(call separator-line,Required Packages)
+	@/usr/bin/pkg list -vH $(REQUIRED_PACKAGES:%=/%)
+	$(call separator-line)
+
+component-environment-prep::
+	@echo "Adding required packages to build environment..."
+	@-echo $(REQUIRED_PACKAGES:%=/%) | xargs \
+		$(PFEXEC) /usr/bin/pkg install --accept -v
```
Comment (alarcher) [inline make-rules/environment.mk]: Then we should document that user needs Software Installation profile.

Comment (xen0l) [inline make-rules/environment.mk]: @alarcher I added it to makefile-targets.txt

File: doc/makefile-targets.txt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+prep::
+  This target is present in all Makefiles.  It can be used at the top level
+  to act across all components, or on an individual component basis.  It is 
+  typically defined by including $(WS_MAKE_RULES)/prep.mk in your component
+  Makefile.  It depends on the previously described "download" target, as well
+  as internal interim targets to unpack source archives and apply patches to
+  the unpacked source.
+
+configure:
```
Comment (pyhalov) [inline doc/makefile-targets.txt]: We don't have it

File: doc/makefile-targets.txt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  of 'manifest-*.published' and clean those up; these can be left over from
+  components that were published but then 'hg rm'd before the next clobber.
+
+REQUIRED_PACKAGES::
+  This target is present in all component Makefiles.  It is automatically
+  defined by including $(WS_MAKE_RULES)/ips.mk.  It is a convenience that can
+  be used to generate the set of REQUIRED_PACKAGES that is needed by the
+  'pkgdepend resolve' portion of package generation and publication (publish
+  target).  It automatically appends to the component Makefile, but the results
+  should be
```
Comment (pyhalov) [inline doc/makefile-targets.txt]: why new line? :)

Comment (xen0l) [inline doc/makefile-targets.txt]: It was taken as it form userland-gate, fixed.

File: doc/makefile-targets.txt (hunk 2 above)
Comment (xen0l) [inline doc/makefile-targets.txt]: removed.

**Pull request merged**